tgui/packages/tgui/interfaces/AirAlarm.tsx

```diff
@@ -0,0 +1,480 @@
+    danger: BooleanLike;
+  }[];
+  thresholdTypeMap: Record<string, number>;
+};
+
+export const AirAlarm = (props, context) => {
+  const { act, data } = useBackend<AirAlarmData>(context);
+  const locked = data.locked && !data.siliconUser;
+  return (
+    <Window width={475} height={650}>
+      <Window.Content scrollable>
+        <InterfaceLockNoticeBox />
+        <AirAlarmStatus />
+        {!locked && <AirAlarmControl />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const AirAlarmStatus = (props, context) => {
+  const { data } = useBackend<AirAlarmData>(context);
+  const dangerMap = {
+    0: {
+      color: 'good',
+      localStatusText: 'Optimal',
+    },
+    1: {
+      color: 'average',
+      localStatusText: 'Caution',
+    },
+    2: {
+      color: 'bad',
+      localStatusText: 'Danger (Internals Required)',
+    },
+  };
+  const localStatus = dangerMap[data.dangerLevel] || dangerMap[0];
+  return (
+    <Section title="Air Status">
+      <LabeledList>
+        {(data.envData.length > 0 && (
+          <>
+            {data.envData.map((entry) => {
+              const status = dangerMap[entry.danger] || dangerMap[0];
+              return (
+                <LabeledList.Item
+                  key={entry.name}
+                  label={entry.name}
+                  color={status.color}>
+                  {entry.value}
+                </LabeledList.Item>
+              );
+            })}
+            <LabeledList.Item label="Local status" color={localStatus.color}>
+              {localStatus.localStatusText}
+            </LabeledList.Item>
+            <LabeledList.Item
+              label="Area status"
+              color={data.atmosAlarm || data.fireAlarm ? 'bad' : 'good'}>
+              {(data.atmosAlarm && 'Atmosphere Alarm') ||
+                (data.fireAlarm && 'Fire Alarm') ||
+                'Nominal'}
+            </LabeledList.Item>
+          </>
+        )) || (
+          <LabeledList.Item label="Warning" color="bad">
+            Cannot obtain air sample for analysis.
+          </LabeledList.Item>
+        )}
+        {!!data.emagged && (
+          <LabeledList.Item label="Warning" color="bad">
+            Safety measures offline. Device may exhibit abnormal behavior.
+          </LabeledList.Item>
+        )}
+      </LabeledList>
+    </Section>
+  );
+};
+
+const AIR_ALARM_ROUTES = {
+  home: {
+    title: 'Air Controls',
+    component: () => AirAlarmControlHome,
+  },
+  vents: {
+    title: 'Vent Controls',
+    component: () => AirAlarmControlVents,
+  },
+  scrubbers: {
+    title: 'Scrubber Controls',
+    component: () => AirAlarmControlScrubbers,
+  },
+  modes: {
+    title: 'Operating Mode',
+    component: () => AirAlarmControlModes,
+  },
+  thresholds: {
+    title: 'Alarm Thresholds',
+    component: () => AirAlarmControlThresholds,
+  },
+};
+
+const AirAlarmControl = (props, context) => {
+  const [screen, setScreen] = useLocalState(context, 'screen', 'home');
```

Comment:
```suggestion
  const [screen] = useLocalState<Screen>(context, 'screen', 'home');
```

using the above type, constrain this to Screen 
we also don't actually need to get the setScreen function from localState unless we're using it

Reply:
![image](https://user-images.githubusercontent.com/54709710/210518126-e4e2407f-7d7c-4d58-a3cc-ea02380add4c.png)
![image](https://user-images.githubusercontent.com/54709710/210518146-be98f954-c823-4da4-976b-ee04b8170691.png)
![image](https://user-images.githubusercontent.com/54709710/210517864-f25d7bb3-1017-4ac4-a67c-c1b3b29b845c.png)

An element array doesn't have the props like key (since its an array) but is still a valid render. I can circumvent this by wrapping the arrays in a fragment, please let me know if this is bad.

Reply:
Wrapping the element array is fine, but I'm still wondering if using route.component() via direct invocation is the best way to do this. It may have no effect at all, but this is otherwise bad practice (a good question for @stylemistake)

https://dev.to/igor_bykov/react-calling-functional-components-as-functions-1d3l

If its possible, try setting the component property to `component: <Smth />`, then we should be able to use `const Component = route.component` or `{route.component}` in the return block

Reply:
Nah we didn't do direct invocation at all, route.component() returns a function, not an element.